### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by vikrantkode**
# Open for review
## Cart Management
      Add To Cart
      Increase/Decrease Quantity
      Price Card Update after increase/decrease cart quantity
      Remove from cart
(CSS is not perfect)

Note : Preview Link is not working So kindly look at the video for UI refrence.

https://user-images.githubusercontent.com/54833758/161104621-b79da6dc-7e7c-4942-a9da-e9f49d3e8d90.mp4

### CODE REVIEW

File: src/components/ProductCard/ProductCard.jsx [hunk 1]
```diff
@@ -1,33 +1,61 @@
...
+import {useCart} from "../../context/Cart-Context"
+import axios from "axios";
+import { useAuth } from "../../context/Auth-Context";
+import { Link, useNavigate } from "react-router-dom";
+
+
+function ProductCard({ item }) {
+
+  const {itemsInCart,setItemsInCart} = useCart();
+  const {state:{encodedToken}} = useAuth();
```
Comment (yashpatel1797) [inline src/components/ProductCard/ProductCard.jsx:12]: if possible you can destructure `encodedToken` in the context itself and pass in value.

File: src/components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -1,33 +1,61 @@
...
+  const navigate = useNavigate();
+
+    const addToCartClickHandler = async (product) => {
+      console.log("clicked here",product)
```
Comment (yashpatel1797) [inline src/components/ProductCard/ProductCard.jsx:16]: remove console from code before PR.

File: src/components/Cart/Cart.jsx [hunk 3]
```diff
@@ -1,54 +1,46 @@
 import React from 'react'
 import "./Cart.css"
 
-function Cart() {
+function Cart({item,cardQtyHandler,removeFromCartHandler}) {
+    console.log("cart card se",item)
```
Comment (Chiragmodi01) [inline src/components/Cart/Cart.jsx:5]: you should remove console logs before pushing.

File: src/components/Cart/Cart.jsx [hunk 4]
```diff
@@ -1,54 +1,46 @@
...
+                    <small className="discount">{item.discount}% off</small>
                 </div>
                 <div className="quantity">
                     <p className="cart_quantity">Quantity: </p>
-                    <button className="quantity_decrease">-</button>
-                    <input className="quantity_box" type="text" />
-                    <button className="quantity_increase">+</button>
+                    {item.qty === 1 ? 
+                    <button disabled="true" className="quantity_decrease" 
+                        onClick={()=>cardQtyHandler(item._id,"decrement")}>-</button> :
```
Comment (Chiragmodi01) [inline src/components/Cart/Cart.jsx:27]: You're using this function multiple times, so better if you extract it out.

File: src/pages/CartPage/CartPage.jsx [hunk 5]
```diff
@@ -1,7 +1,70 @@
 import React from "react";
+import "./CartPage.css";
+import Nav from "../../components/Nav/Nav";
```
Comment (yashpatel1797) [inline src/pages/CartPage/CartPage.jsx:3]: you can create `index.js` file for folders and export components from there.

**Review by yashpatel1797 — APPROVED** [review]
nice file structure.
code looks clean.
added few suggestions.

File: src/pages/Auth/LoginPage/LoginPage.jsx [hunk 6]
```diff
@@ -34,7 +48,7 @@ const LoginPage = () => {
         <Nav />
         <div className="main_container">
       <div className="form_container">
-        <form className="form" onSubmit={ loginClickHandler }>
+        <form className="form" >
```
Comment (Chiragmodi01) [inline src/pages/Auth/LoginPage/LoginPage.jsx:51]: className could be improved like 'login-form', looks pretty generic now.

File: src/pages/Auth/LoginPage/LoginPage.jsx [hunk 7]
```diff
@@ -64,17 +78,17 @@ const LoginPage = () => {
             <li className="normal_list login_list">
               <label><input className="checkbox_list" type="checkbox" />
               Remember Me</label>
-              <Link to="/#" className="login_form_link forgot_pwd_link" >
+              <Link  to=""className="login_form_link forgot_pwd_link" >
```
Comment (Chiragmodi01) [inline src/pages/Auth/LoginPage/LoginPage.jsx:81]: 'to' missing!

Comment (vikrantkode) [reply src/pages/Auth/LoginPage/LoginPage.jsx:81]: coz forgot password page is not there. So I haven't added that

File: src/pages/Auth/LoginPage/LoginPage.jsx (hunk 6 above)
Comment (vikrantkode) [reply src/pages/Auth/LoginPage/LoginPage.jsx:51]: okay. Will change it

File: src/components/Cart/Cart.jsx (hunk 4 above)
Comment (vikrantkode) [reply src/components/Cart/Cart.jsx:27]: I didnt understood.. Can you please elaborate a little bit ?

File: src/components/Cart/Cart.jsx (hunk 3 above)
Comment (vikrantkode) [reply src/components/Cart/Cart.jsx:5]: I forgot to remove from here. Will remove it

File: src/pages/CartPage/CartPage.jsx (hunk 5 above)
Comment (vikrantkode) [reply src/pages/CartPage/CartPage.jsx:3]: Sure. After making all i will move it to index.js file

File: src/components/ProductCard/ProductCard.jsx (hunk 2 above)
Comment (vikrantkode) [reply src/components/ProductCard/ProductCard.jsx:16]: Sure. I forgot to remove

File: src/components/ProductCard/ProductCard.jsx (hunk 1 above)
Comment (vikrantkode) [reply src/components/ProductCard/ProductCard.jsx:12]: in Context its destructured

**Pull request merged**